server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputerTests.java

```diff
@@ -683,7 +716,7 @@ public void testDesiredBalanceShouldConvergeInABigCluster() {
                 "Failed to converge desired balance for [{}] nodes and [{}] total shards:\n {}",
                 nodes,
                 totalShards,
-                clusterState.getRoutingNodes()
+                clusterState.getRoutingNodes().toString()
```

Comment:
Logger has special handling for iterables and without to string it only outputs toString for every RoutingNode rather then `RoutingNodes#toString()`. The latter contains mode details

Reply:
++ what a mess :)